File: packages/ios/Avocado/Avocado/Plugins/Browser.swift
import Foundation
import SafariServices

public class Browser : Plugin, SFSafariViewControllerDelegate {
  var vc: SFSafariViewController?
  
  @objc func open(_ call: PluginCall) {
    if let urlString = call.options["url"] as? String {
      let url = URL(string: urlString)
      vc = SFSafariViewController.init(url: url!)
      vc!.delegate = self
      avocado.viewController.present(vc!, animated: true, completion: {
        
      })
    }
  }
}

File: packages/js/src/plugins/browser.ts
import { NativePlugin, Plugin } from '../plugin';


@NativePlugin({
  name: 'Browser',
  id: 'com.avocadojs.plugin.browser'
})
export class Browser extends Plugin {

  open(url: string) {
    if (this.isNative) {
      this.nativeCallback('open', { url });
    }

    window.open(url);
  }

}
